source/code/plugins/oms_common.rb

```diff
@@ -575,34 +712,22 @@ def get_os_version(conf_path = "/etc/opt/microsoft/scx/conf/scx-release")
         return @@OSVersion
       end
 
-      def get_hostname
-        return @@Hostname if !@@Hostname.nil?
+      def get_hostname(ignoreOldValue=false)
+        # Notes:  This implementation does not try to address multi-byte character aspects,
+        # but merely tries to address simple exigencies.  There may be multi-byte issues
+        # we will need to address in the future here.
+        if ! is_hostname_compliant?(@@Hostname) or ignoreOldValue then
```

Comment:
style nit: place ! directly next to is_hostname_compliant?: `!is_hostname_compliant?(@@Hostname)`

Reply:
I put a "not".